src/Parsers/ASTColumnsTransformers.cpp

```diff
@@ -151,9 +154,15 @@ void ASTColumnsReplaceTransformer::transform(ASTs & nodes) const
                 replaceChildren(new_ast, column, replace_it->first);
                 column = new_ast;
                 column->setAlias(replace_it->first);
+                ++replace_column_sucess;
             }
```

Comment:
You can add a flag here to check whether a replace takes effect, and throw an error with the related AST right after.